front/src/modules/timetableItem/components/ManageTimetableItem/TrainSettings.tsx

```diff
@@ -140,8 +113,14 @@ const TrainSettings = () => {
               <small className="text-nowrap">{t('category')}</small>
             </>
           }
-          onChange={setTrainCategory}
-          value={trainCategory ?? { label: tRollingStock('rollingStock.categoriesOptions.choose') }}
+          onChange={(option) => {
+            if (option && 'id' in option && option.category) {
+              dispatch(updateCategory(option.category));
+            } else {
+              dispatch(updateCategory(null));
+            }
```

Comment:
`options` is of type `CategoryOptionWithId | undefined`
```suggestion
            if (option !== undefined) {
              dispatch(updateCategory(option?.category));
            }
```

Reply:
Done

Reply:
Forgot in the suggestion, but we can also remove the `?` l.120